packages/amplication-server/src/core/resource/resourceBtm.service.ts

```diff
@@ -165,16 +160,20 @@ export class ResourceBtmService {
       };
     }
 
-    const recommendations = await this.prepareBtmRecommendations(
-      JSON.stringify(metadata),
-      resourceId
-    );
+    try {
+      const recommendations = await this.prepareBtmRecommendations(
+        metadata.data,
+        resourceId
+      );
 
-    return {
-      status: EnumUserActionStatus.Completed,
-      originalResourceId: resourceId,
-      data: recommendations,
-    };
+      return {
+        status: EnumUserActionStatus.Completed,
+        originalResourceId: resourceId,
+        data: recommendations,
+      };
+    } catch (error) {
+      throw new AmplicationError(BREAK_THE_MONOLITH_AI_ERROR_MESSAGE);
```

Comment:
Let's log the original `error` object here to our login infra so we don't lose it

Reply:
done